processor/partition_worker.go

```diff
@@ -0,0 +1,145 @@
+package processor
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/services/rsources"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type partitionWorker struct {
+	partition string
+	workers   []*pipelineWorker
```

Comment:
minor, just a suggestion
```suggestion
	pipelines   []*pipelineWorker
```